bulk_processing/new_address_processor.py

```diff
@@ -0,0 +1,94 @@
+        'SPG_VDNEW'}
+
+    ESTAB_TYPES = {'HALL OF RESIDENCE', 'CARE HOME', 'HOSPITAL', 'HOSPICE', 'MENTAL HEALTH HOSPITAL',
+                   'MEDICAL CARE OTHER', 'BOARDING SCHOOL', 'LOW/MEDIUM SECURE MENTAL HEALTH',
+                   'HIGH SECURE MENTAL HEALTH', 'HOTEL', 'YOUTH HOSTEL', 'HOSTEL', 'MILITARY SLA', 'MILITARY US',
+                   'RELIGIOUS COMMUNITY', 'RESIDENTIAL CHILDRENS HOME', 'EDUCATION OTHER', 'PRISON',
+                   'IMMIGRATION REMOVAL CENTRE', 'APPROVED PREMISES', 'ROUGH SLEEPER', 'STAFF ACCOMMODATION',
+                   'CAMPHILL', 'HOLIDAY PARK', 'HOUSEHOLD', 'SHELTERED ACCOMMODATION', 'RESIDENTIAL CARAVAN',
+                   'RESIDENTIAL BOAT', 'GATED APARTMENTS', 'MOD HOUSEHOLDS', 'FOREIGN OFFICES', 'CASTLES', 'GRT SITE',
+                   'MILITARY SFA', 'EMBASSY', 'ROYAL HOUSEHOLD', 'CARAVAN SITE', 'MARINA', 'TRAVELLING PERSONS',
+                   'TRANSIENT PERSONS'}
+
+    file_prefix = Config.BULK_NEW_ADDRESS_FILE_PREFIX
+    routing_key = Config.NEW_ADDRESS_EVENT_ROUTING_KEY
+    exchange = ''
+    bucket_name = Config.BULK_NEW_ADDRESS_BUCKET_NAME
+    project_id = Config.BULK_NEW_ADDRESS_PROJECT_ID
+    schema = {
+
+        'UPRN': [mandatory(), max_length(12), numeric(), no_padding_whitespace()],
+        'ESTAB_UPRN': [mandatory(), max_length(12), numeric(), mandatory(), no_padding_whitespace()],
+        'ADDRESS_TYPE': [mandatory(), in_set({'HH', 'CE', 'SPG'}, label='address_types'), no_padding_whitespace()],
+        'ESTAB_TYPE': [mandatory(), in_set(ESTAB_TYPES, label='estab types'), max_length(255), no_padding_whitespace()],
+        'ADDRESS_LEVEL': [mandatory(), in_set({'E', 'U'}, label='address levels'), no_padding_whitespace()],
+        'ABP_CODE': [mandatory(), max_length(6), no_padding_whitespace()],
+        'ORGANISATION_NAME': [max_length(60), no_padding_whitespace()],
+        'ADDRESS_LINE1': [mandatory(), max_length(60), no_padding_whitespace()],
+        'ADDRESS_LINE2': [max_length(60), no_padding_whitespace()],
+        'ADDRESS_LINE3': [max_length(60), no_padding_whitespace()],
+        'TOWN_NAME': [mandatory(), max_length(30), no_padding_whitespace()],
+        'POSTCODE': [mandatory(), max_length(8), no_padding_whitespace()],
+        'LATITUDE': [mandatory(), latitude_longitude(max_scale=7, max_precision=9), no_padding_whitespace()],
+        'LONGITUDE': [mandatory(), latitude_longitude(max_scale=7, max_precision=10), no_padding_whitespace()],
+        'OA': [mandatory(), max_length(9), no_padding_whitespace()],
+        'LSOA': [mandatory(), max_length(9), no_padding_whitespace()],
+        'MSOA': [mandatory(), max_length(9), no_padding_whitespace()],
+        'LAD': [mandatory(), max_length(9), no_padding_whitespace()],
+        'REGION': [mandatory(), max_length(9), no_padding_whitespace(), region_matches_treatment_code()],
+        'HTC_WILLINGNESS': [mandatory(), in_set({'0', '1', '2', '3', '4', '5'}, label='HTC_WILLINGNESS'),
```

Comment:
We should decide on one convention for the labels since they get passed into the error details we feed back. Using the column header makes sense to me.